libvast/vast/concept/parseable/core/parser.hpp

```diff
@@ -130,7 +119,7 @@ template <class T>
 using is_parser_t = typename is_parser<T>::type;
 
 template <class T>
-constexpr bool is_parser_v = std::is_base_of<parser<T>, T>::value;
+concept is_parser_v = is_parser<std::decay_t<T>>::value;
```

Comment:
```suggestion
concept parser = is_parser<std::decay_t<T>>::value;
```

Would this be possible, or be name conflict? (Same for `printer`.)

Reply:
Good thought -- it would be a name conflict. I was planning on renaming some things in a follow-up PR. E.g. change the `parser` class type to be `parser_t` and then rename the concept `is_parser_v` to `parser`. Same with `printer`. I thought that would introduce more noise than value for review purposes of this PR.

Reply:
Nice idea. I agree that the renaming could be a bit noisy in here. Although in general, with clear review instructions, you can make the reviewer look at your PR the way you want...most of the time. 😉 

Regarding the concrete renaming: how about we use something like `basic_parser` or `parser_base`? It's a bit more semantic that `*_t`.

Reply:
I don't feel strongly on the rename. I like `parser_base` too, so we can go with that.